src/WorkflowManager/Database/Repositories/TaskExecutionStatsRepository.cs

```diff
@@ -187,66 +181,86 @@ private static ExecutionStats ExposeExecutionStats(ExecutionStats taskExecutionS
+        private static void CalculatePodExecutionTime(ExecutionStats taskExecutionStats, TaskExecution taskUpdateEvent, IEnumerable<string> statKeys)
         {
-            var statusNull = string.IsNullOrWhiteSpace(status);
-            var workflowNull = string.IsNullOrWhiteSpace(workflowId);
-            var taskIdNull = string.IsNullOrWhiteSpace(taskId);
+            var start = DateTime.Now;
+            var end = new DateTime();
+            foreach (var statKey in statKeys)
+            {
+                if (statKey.Contains("StartTime") && DateTime.TryParse(taskUpdateEvent.ExecutionStats[statKey], out var startTime))
+                {
+                    start = (startTime < start ? startTime : start);
+                }
+                else if (DateTime.TryParse(taskUpdateEvent.ExecutionStats[statKey], out var endTime))
+                {
+                    end = (endTime > end ? endTime : start);
+                }
+            }
+            taskExecutionStats.ExecutionTimeSeconds = (end - start).TotalMilliseconds / 1000;
+        }
 
-            return await _taskExecutionStatsCollection.CountDocumentsAsync(T =>
-            T.StartedUTC >= start.ToUniversalTime() &&
-            T.StartedUTC <= endTime.ToUniversalTime() &&
-            (workflowNull || T.WorkflowId == workflowId) &&
-            (taskIdNull || T.TaskId == taskId) &&
-            (statusNull || T.Status == status));
+        public async Task<long> GetStatsStatusCountAsync(DateTime startTime, DateTime endTime, string status = "", string workflowId = "", string taskId = "")
+        {
+            Expression<Func<ExecutionStats, bool>>? statusFilter = null;
+            if (!string.IsNullOrWhiteSpace(status))
+            {
+                statusFilter = t => t.Status == status;
+            }
+            return await GetStatsCountAsync(startTime, endTime, statusFilter, workflowId, taskId);
         }
 
-        public async Task<long> GetStatsStatusSucceededCountAsync(DateTime startTime, DateTime endTime, string workflowId = "", string taskId = "")
+        public async Task<long> GetStatsCountAsync(DateTime startTime, DateTime endTime, Expression<Func<ExecutionStats, bool>>? statusFilter = null, string workflowId = "", string taskId = "")
         {
             var workflowNull = string.IsNullOrWhiteSpace(workflowId);
             var taskIdNull = string.IsNullOrWhiteSpace(taskId);
 
-            return await _taskExecutionStatsCollection.CountDocumentsAsync(T =>
-            T.StartedUTC >= startTime.ToUniversalTime() &&
-            T.StartedUTC <= endTime.ToUniversalTime() &&
-            (workflowNull || T.WorkflowId == workflowId) &&
-            (taskIdNull || T.TaskId == taskId) &&
-            T.Status == TaskExecutionStatus.Succeeded.ToString());
+            var builder = Builders<ExecutionStats>.Filter;
+            var filter = builder.Empty;
+
+            filter &= builder.Where(t => t.StartedUTC == startTime.ToUniversalTime());
```

Comment:
This doesn't look right?

Reply:
![image](https://github.com/Project-MONAI/monai-deploy-workflow-manager/assets/61380713/bda7f27f-9c9c-433e-94c8-6ce5566f7385)

changes start and end time to be >= & <=

Reply:
yeah looks like it should be 
t => t.StartedUTC >= startTime.ToUniversalTime());
and
t => t.StartedUTC <= endTime.ToUniversalTime());